vizro-core/src/vizro/models/_base.py

```diff
@@ -101,6 +218,72 @@ def _is_discriminated_union(field):
         cls.update_forward_refs(**vm.__dict__.copy())
         new_type.update_forward_refs(**vm.__dict__.copy())
 
+    def dict(self, **kwargs):
+        # Overwrite pydantic's own `dict` method to add __vizro_model__ to the dictionary.
+        # To get exclude as an argument is a bit fiddly because this function is called recursively
+        # inside pydantic, which sets exclude=None by default.
+        if kwargs.get("exclude") is None:
+            kwargs["exclude"] = self.__vizro_exclude_fields__()
+        _dict = super().dict(**kwargs)
+        # Inject __vizro_model__ into the dictionary - needed just for to_python.
+        _dict["__vizro_model__"] = self.__class__.__name__
```

Comment:
Curious why `__vizro_model__` cannot be directly defined in each Vizro model. 

In vizro-ai, the card create is using `dict` in the flow of `Card` -> `Card.dict()` -> `replace with a readable id` -> `Card.parse_obj(proxy_dict)`, which would fail validation as 
```
ValidationError: 1 validation error for Card
__vizro_model__
  extra fields not permitted (type=value_error.extra)
  ```
  
it can be fixed by adding `proxy_dict.pop("__vizro_model__", None)`, but also wondering how often would this cause error, when a user would like to do ".dict()" then ".parse_obj()" (I guess not too often)

Reply:
This is a good point! And something I had not considered as such. In a previous iteration it did exist as a field, but @antonymilne preferred not to add it as such. Curious to see what he thinks!

Reply:
For our case a couple of additional thoughts:
- we could also use the `exclude` argument of `dict()` as an alternative fix
- why do we go this route again? Is it important to have a readable ID? I'd prefer to not user proxies unless needed (like for the filter control)

Reply:
Very good point indeed @lingyielia.

I think the best solution is to make our overriding of `dict` less disruptive like this:
```
        # Dunder properties like __vizro_model__ are ignored even when explicitly in `include` so we need to add this
        # manually.
        if kwargs.get("include") is not None and "__vizro_model__" in kwargs["include"]:
            _dict["__vizro_model__"] = self.__class__.__name__
```

And then in `_to_python` we do `dict(include={"__vizro_model__"})` to explicitly include it.

This way the default `dict` behaviour is left the same as the pydantic default which feels better.

Reply:
P.S. probably worth adding a new test to ensure that the basic `.dict()` functionality remains unaltered i.e. does not include `__vizro_model__`. So we don't accidentally break this in future.

Reply:
> * Is it important to have a readable ID?

Yes, right now all components (graph, table, and card) have readable IDs, which will be used by filter and layout creation.

Reply:
I like the idea "basic .dict() functionality remains unaltered" so it's not disruptive in any possible cases 🙌

Reply:
ressolved now by using global `_PATCH_VIZRO_BASE_MODEL_DICT`